lib/game_team.rb

```diff
@@ -146,6 +186,44 @@ def self.least_goals_scored(team_id)
     return min_goals[1]["goals"]
   end
 
+  def self.opponents_records(team_id)
+    game_ids = all.map {|gt| gt.game_id if gt.team_id == team_id }.compact
+    opponents_records = Hash.new { |hash, key| hash[key] = [] }
+    game_ids.each do |game_id|
+      all.find_all do |gt|
+        opponents_records[gt.team_id] << gt.result if gt.game_id == game_id && gt.team_id != team_id
+      end
+    end
+    opponents_records
```

Comment:
This leaves me with a hash with the team_id as the key and an array as the value that contains their results(["WIN", "LOSS", "TIE"])

Reply:
This would be a great candidate for a nested hash